lib/core/utils/qsa.js

```diff
@@ -215,10 +210,19 @@ matchExpressions = function  (domTree, expressions, recurse) {
 	}, []);
 };
 
+/**
+ * querySelectorAll implementation that works on the virtual DOM and supports shadowDOM
```

Comment:
change this to `querySelectorAll implementation that operates on the flattened tree (supports shadow DOM)`